### DISCUSSION TITLE
Pull request improvement needed

### DISCUSSION THREAD

**Discussion by Aswinmcw** [community/community#120864] [Pull Requests]
### Select Topic Area

Bug

### Body

In current GitHub following issue is there, Consider I have created a pull request in a repo and code owner approved the changes, then during rebase before merging, if I change the approved files, it will allow me to merge without any issues as it has approval without the knowledge of approved person

My suggestion is that if any approved file changed, then it will be better if I comment is added in the PR mentioning that the approved files changed

In addition to that we can add a setting where the repo owner can turn it on so that if approved files are changed then it will automatically re request the code owners blocking the user from merging

**Comment by Kolass2004** [comment]
Adding a comment in the pull request mentioning that approved files have been changed sounds like a helpful notification to maintain transparency. Implementing a setting for repo owners to automatically re-request code owner review when approved files are changed could enhance control over the merging process and ensure accountability. It adds an extra layer of validation before merging changes.

**Comment by Aswinmcw** [comment]
Can it be implemented?
